flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecOverAggregate.java

```diff
@@ -183,15 +190,15 @@ protected Transformation<RowData> translateToPlanInternal(
 
         final int orderKey = orderKeys[0];
         final LogicalType orderKeyType = inputRowType.getFields().get(orderKey).getType();
-        // check time field && identify window rowtime attribute
-        final int rowTimeIdx;
+        // check time field && identify window time attribute
+        TimeAttribute timeAttribute;
         if (isRowtimeAttribute(orderKeyType)) {
-            rowTimeIdx = orderKey;
+            timeAttribute = TimeAttribute.ROW_TIME;
         } else if (isProctimeAttribute(orderKeyType)) {
-            rowTimeIdx = -1;
+            timeAttribute = TimeAttribute.PROC_TIME;
         } else {
-            throw new TableException(
-                    "OVER windows' ordering in stream mode must be defined on a time attribute.");
+            timeAttribute = TimeAttribute.NON_TIME;
+            LOG.info("Non-time attribute window detected");
```

Comment:
It's uncommon that we log this tiny properties within the planner. I recommend to drop it.

Reply:
Removed the extraneous log line